testing/gyp: upgrade to 20190716

OTLabs opened the pull request:
- Add cleanup_srcdir()

OTLabs commented:
@maxice8 could you please review proposed upgrade and changes? Thank you!

maxice8 commented:
Can you check that stuff like telegram-desktop still work ? I also did a gyp to python3 conversion once but telegram-desktop failed to build

OTLabs commented:
I will check the possibility to switch to python3 on telegram-desktop, thank you for a tip. Meanwhile I have refactored the PR to update gyp to latest version and added cleanup_srcdir() and removed switch to python3.

JFYI, I was able to run all tests except this one: test/generator-output/gyptest-actions.py

maxice8 commented on testing/gyp/APKBUILD:
```diff
@@ -21,4 +21,9 @@ package() {
 
-sha512sums="bfe8484290e7216888c001638a5b6e42fc254ecf13320c9398a1c2a71e968f8338aebe2fe6edc64de3fe75203ad704bd8976caef21affe21f97430b838fefc52  aca1e2c3d346d704adfa60944e6b4dd06f4728be.tar.gz"
+cleanup_srcdir() {
+	[ -d src ] && chmod -R +w src
```
Any reason why this is necessary ? as far as i am aware the only ones that need it are go packages that use modules because thay are made read-only.

OTLabs replied:
For unknown reason I got same problem as you mention for go packages during several unsuccessful builds of this package. I have not dig to much in this issue, just solve it this way.
